Add `engine` property in `package.json`

Some breaking changes in the `rescript` binary node may not be captured. We can add the [`engines`](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#engines) property with `engine-strict` to catch errors on CI.

The `format -check` command (https://github.com/rescript-lang/rescript-compiler/pull/5760) introduced in v10.1.0 broke the requirements described in the [documentation](https://rescript-lang.org/docs/manual/latest/installation#prerequisites).

> [Node.js](https://nodejs.org/) version >= 10

Issue: https://github.com/rescript-association/rescript-lang.org/pull/675

Example:

package.json:
```json
{
  "engineStrict": true,
  "engines": {
    "node": ">=18"
  }
} 
```

Run on CI (node v16):

```
npm i rescript-*.tgz --engine-strict
```

```
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: rescript@11.0.0-alpha.5
npm ERR! notsup Not compatible with your version of node/npm: rescript@11.0.0-alpha.5
npm ERR! notsup Required: {"node":">=18"}
npm ERR! notsup Actual:   {"npm":"8.19.3","node":"v16.19.1"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pedro/.npm/_logs/2023-04-20T20_04_38_734Z-debug-0.log

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,10 @@
 
 - Parse `assert` as a regular function. `assert` is no longer a unary expression. Example: before `assert 1 == 2` is parsed as `(assert 1) == 2`, now it is parsed as `assert(1 == 2)`. https://github.com/rescript-lang/rescript-compiler/pull/6180
 
+#### :bug: Bug Fix
+
+- Make "rescript format" work with node 10 again and set minimum required node version to 10 in package.json. https://github.com/rescript-lang/rescript-compiler/pull/6186
+
 # 11.0.0-alpha.4
 
 #### :rocket: Main New Feature

--- a/package.json
+++ b/package.json
@@ -7,6 +7,9 @@
     "prettier": "^2.7.1",
     "rollup": "^0.49.2"
   },
+  "engines": {
+    "node": ">=10"
+  },
   "bin": {
     "bsc": "bsc",
     "bstracing": "lib/bstracing",

--- a/scripts/rescript_format.js
+++ b/scripts/rescript_format.js
@@ -5,7 +5,7 @@ var format_usage = `Usage: rescript format <options> [files]
 \`rescript format\` formats the current directory
 `;
 var child_process = require("child_process");
-var util = require("node:util");
+var util = require("util");
 var asyncExecFile = util.promisify(child_process.execFile);
 var path = require("path");
 var fs = require("fs");

